tools/ci_build/build.py

```diff
@@ -376,6 +376,33 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
     if is_windows():
         cmake_args += cmake_extra_args
 
+    # ADO pipelines will store the pipeline build number (e.g. 191101-2300.1.master) and 
+    # source version in environment variables. If present, use these values to define the 
+    # WinML/DML DLL versions.
+    build_number = os.getenv('Build_BuildNumber')
+    source_version = os.getenv('Build_SourceVersion')
+    if build_number and source_version:
+        print('found env vars')
+        matches = re.match(r"^(\d\d)(\d\d)(\d\d)-(\d\d)(\d\d)\.(\d)\.(\S+)$", build_number)
+        if matches:
+            print('matched build number')
+            YY = matches.group(1)
+            MM = matches.group(2)
+            DD = matches.group(3)
+            HH = matches.group(4)
+
+            # Example (BuildNumber: 191101-2300.1.master, SourceVersion: 0bce7ae6755c792eda558e5d27ded701707dc404)
+            # MajorPart = 1
+            # MinorPart = 0
+            # BuildPart = 1911
+            # PrivatePart = 123
+            # String = 191101-2300.1.master.0bce7ae
+            cmake_args += ["-DWINML_VERSION_MAJOR_PART=1",
```

Comment:
Would it be better to get the major and minor part from https://github.com/microsoft/onnxruntime/blob/master/VERSION_NUMBER ?

Reply:
For context, @pranavsharma mentioned via email that it is okay for the major and minor part to match that of the winml binary. It is also useful of the two to match for auditing purposes, but that being said the build information would be enough information to match up the dlls. I'll go ahead and change this.